### PR TITLE
#80240 Integrate Sign-in into ARP

### PR DESCRIPTION
**Pull request by gabezurita**
##  Demo
[Watch this demo showing sign-in working with the ARP engine endpoint!](https://www.loom.com/share/1189bde814794a9997e75d5c2384fe66?sid=6a442864-9682-45ce-9f8c-8a4842411760)

## Accompanying PR
https://github.com/department-of-veterans-affairs/vets-api/pull/17010

## Description
Integrates Accredited Representative Portal (ARP) sign-in functionality using the React Router and Redux libraries. Key changes include:

- Populate a user in client state at root component load using redux
  - use accredited_representative_portal/v0/user
- Update header user navigation component and landing page
  - Renders loading, logged-in/out states with sign-in/out links
  - Uses /v0/representative_user endpoint to populate client state
- Gate POA table based on signed-in status
  - conditionally renders based loading, not signed-in, and signed-in states
- Integrate feature toggles directly into StartReactApp within ARP

The sign-in integration allows accredited representatives to authenticate and access the portal. The user's logged-in state is tracked in the Redux store. The header reflects the current auth state and provides sign-in/sign-out links.

## Related issue(s)
- [#80240 Integrate Sign-in into ARP](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/80240)
- Follow-on issues
    - [84526](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/84526) - Create an ARP mock user in [vets-api-mock-data](https://github.com/department-of-veterans-affairs/vets-api-mockdata) to test/demo ARP login in staging and localhost
    - [84531](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/84531) - Handle [id.me](http://id.me/) and [login.gov](http://login.gov/) service status in case of [downtime — see example](https://github.com/department-of-veterans-affairs/vets-website/blob/b57203217ce6758b4517452a85d4dff89cd949fe/src/platform/user/authentication/components/LoginHeader.jsx#L21)
    - [84533](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/84533) - Log users out on inactivity ([session timeout](https://github.com/department-of-veterans-affairs/vets-website/blob/374bd5cc7221d39121c91139e5eac79fc51c8afb/src/platform/user/authentication/components/SessionTimeoutModal.jsx#L1-L160))

## Testing done

- see unit and e2e tests.
- more thorough, manual testing will be done once ARP has a mock user we can use on localhost

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Comment by gabezurita** [comment]
@amprokop, @williamphelps13, and @nihil2501, a review on this would be appreciated!

**Comment by gabezurita** [comment]
@department-of-veterans-affairs/octo-identity, would you like to review this front-end PR, too? More work must be done (see PR description), but it's a start.

File: src/applications/accredited-representative-portal/actions/user.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+export function fetchUser() {
+  return async dispatch => {
+    dispatch({ type: FETCH_USER });
+    try {
+      const response = await apiRequest(
+        `${environment.API_URL}/accredited_representative_portal/v0/user`,
+        {
+          method: 'GET',
+          // credentials: 'include', // TODOARF: Do we need this?
```
[the remark this answers is not in the document]
Comment (gabezurita) [reply src/applications/accredited-representative-portal/actions/user.js]: See commit uncommenting this out:
fcd8ed8d32

[the remark this answers is not in the document]
Comment (gabezurita) [reply src/applications/accredited-representative-portal/actions/user.js]: I also made a [demo showing sign-in working with the ARP engine endpoint!](https://www.loom.com/share/1189bde814794a9997e75d5c2384fe66?sid=6a442864-9682-45ce-9f8c-8a4842411760) if you want to take a look 😄

**Review by williamphelps13 — CHANGES_REQUESTED** [review]
Appreciate the demo video as well!

File: src/applications/accredited-representative-portal/containers/App.jsx [hunk 2]
```diff
@@ -1,26 +1,40 @@
...
+  const isAppToggleLoading = useToggleLoadingValue(
+    TOGGLE_NAMES.accreditedRepresentativePortalFrontend,
+  );
+
   const isAppEnabled = useToggleValue(
     TOGGLE_NAMES.accreditedRepresentativePortalFrontend,
   );
 
-  const isProduction = environment.isProduction();
+  const isProduction = window.Cypress || environment.isProduction();
```
[the remark this answers is not in the document]
Comment (gabezurita) [reply src/applications/accredited-representative-portal/containers/App.jsx:39]: It's necessary for simulating production in e2e testing. Feel free to play with it!

File: src/applications/accredited-representative-portal/containers/App.jsx [hunk 3]
```diff
@@ -31,20 +45,20 @@ function App() {
...
+    document.location.replace('/');
+    return null;
   }
-
   return (
     <>
       <Header />
-      <Outlet />
+      {isLoading ? (
+        <VaLoadingIndicator message="Loading user information (App)..." />
```
[the remark this answers is not in the document]
Comment (gabezurita) [reply src/applications/accredited-representative-portal/containers/App.jsx]: Done, see commit: f875e303a0

File: src/applications/accredited-representative-portal/tests/components/common/Header/MobileHeader/MobileLogoRow.unit.spec.js [hunk 4]
```diff
@@ -1,34 +1,58 @@
 import React from 'react';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
 import { render } from '@testing-library/react';
 import { expect } from 'chai';
 import { MemoryRouter } from 'react-router-dom-v5-compat';
 
-import MobileLogoRow from '../../../../../components/common/Header/MobileHeader/MobileLogoRow';
+import { MobileLogoRow } from '../../../../../components/common/Header/MobileHeader/MobileLogoRow';
```
[the remark this answers is not in the document]
Comment (gabezurita) [reply src/applications/accredited-representative-portal/tests/components/common/Header/MobileHeader/MobileLogoRow.unit.spec.js:8]: See commit: https://github.com/department-of-veterans-affairs/vets-website/commit/f875e303a0cd04c45cea5972098cd5571ba4b463

File: src/applications/accredited-representative-portal/components/common/Header/Header.jsx [hunk 5]
```diff
@@ -1,8 +1,6 @@
 import React from 'react';
```
[the remark this answers is not in the document]
Comment (gabezurita) [reply src/applications/accredited-representative-portal/components/common/Header/Header.jsx]: Yep, see commit: https://github.com/department-of-veterans-affairs/vets-website/commit/f875e303a0cd04c45cea5972098cd5571ba4b463

File: src/applications/accredited-representative-portal/tests/components/common/Header/MobileHeader/MobileLogoRow.unit.spec.js [hunk 6]
```diff
@@ -1,34 +1,58 @@
...
+    </Provider>,
+  );
+};
+
+describe('MobileLogoRow', () => {
+  it('should render the logo with correct alt text and source', () => {
+    const { getByTestId } = renderMobileLogoRow(false, null);
+    const logo = getByTestId('mobile-logo-row-logo');
+    expect(logo).to.exist;
+    expect(logo.alt).to.equal(
```
[the remark this answers is not in the document]
Comment (gabezurita) [reply src/applications/accredited-representative-portal/tests/components/common/Header/MobileHeader/MobileLogoRow.unit.spec.js]: Yep, see commit: https://github.com/department-of-veterans-affairs/vets-website/commit/f875e303a0cd04c45cea5972098cd5571ba4b463

File: src/applications/accredited-representative-portal/actions/user.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import { apiRequest } from '@department-of-veterans-affairs/platform-utilities/api';
+import environment from '~/platform/utilities/environment';
+
+export const FETCH_USER = 'FETCH_USER';
+export const FETCH_USER_SUCCESS = 'FETCH_USER_SUCCESS';
+export const FETCH_USER_FAILURE = 'FETCH_USER_FAILURE';
+
+export function fetchUser() {
```
[the remark this answers is not in the document]
Comment (gabezurita) [reply src/applications/accredited-representative-portal/actions/user.js]: Sure, why not update all the non-arrow functions in the app:
https://github.com/department-of-veterans-affairs/vets-website/commit/f875e303a0cd04c45cea5972098cd5571ba4b463

File: src/applications/accredited-representative-portal/components/common/Header/common/UserNav.jsx [hunk 8]
```diff
@@ -1,13 +1,21 @@
...
+  selectUserProfile,
+  selectUserIsLoading,
+} from '../../../../selectors/user';
 
-const testProfile = null;
-// const testProfile = { firstName: 'James', lastName: 'Smith' };
+const generateUniqueId = () =>
+  `account-menu-${Math.random()
+    .toString(36)
+    .substr(2, 9)}`;
```
[the remark this answers is not in the document]
Comment (gabezurita) [reply src/applications/accredited-representative-portal/components/common/Header/common/UserNav.jsx]: Yep, see commit: https://github.com/department-of-veterans-affairs/vets-website/commit/f875e303a0cd04c45cea5972098cd5571ba4b463

**Review by gabezurita — COMMENTED** [review]
Addressed all feedback in the below commit:
https://github.com/department-of-veterans-affairs/vets-website/commit/f875e303a0cd04c45cea5972098cd5571ba4b463

Thanks for the thorough review! Agreed with all the style/nit/preference changes not part of the VA lint rules.

**Review by CBonade — DISMISSED** [review]
Approved! Looks like there's a conflict that will need resolving before merge, though.

File: src/applications/accredited-representative-portal/actions/user.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      const response = await apiRequest(
+        `${environment.API_URL}/accredited_representative_portal/v0/user`,
+        {
+          method: 'GET',
+          credentials: 'include',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        },
+      );
```
[the remark this answers is not in the document]
Comment (gabezurita) [reply src/applications/accredited-representative-portal/actions/user.js]: Had the change stashed! https://github.com/department-of-veterans-affairs/vets-website/pull/30146/commits/0c7cc4985fbdac5b6c7b4d5d37342c76f1229f89

File: src/applications/accredited-representative-portal/tests/components/common/Header/MobileHeader/MobileLogoRow.unit.spec.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (gabezurita) [reply src/applications/accredited-representative-portal/tests/components/common/Header/MobileHeader/MobileLogoRow.unit.spec.js]: Had the change stashed! https://github.com/department-of-veterans-affairs/vets-website/pull/30146/commits/0c7cc4985fbdac5b6c7b4d5d37342c76f1229f89

File: src/applications/accredited-representative-portal/actions/user.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+        dispatch({
+          type: FETCH_USER_SUCCESS,
+          payload: profile,
+        });
+      } else {
+        dispatch({ type: FETCH_USER_FAILURE });
+      }
+    } catch (error) {
+      dispatch({ type: FETCH_USER_FAILURE });
+    }
```
[the remark this answers is not in the document]
Comment (gabezurita) [reply src/applications/accredited-representative-portal/actions/user.js]: Had the change stashed! https://github.com/department-of-veterans-affairs/vets-website/pull/30146/commits/0c7cc4985fbdac5b6c7b4d5d37342c76f1229f89

File: src/applications/accredited-representative-portal/components/LandingPageWelcome/LandingPageWelcome.jsx [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+);
+
+LandingPageWelcome.propTypes = {
+  children: PropTypes.node,
+  firstName: PropTypes.string,
+};
+
+LandingPageWelcome.defaultProps = {
+  firstName: '',
+};
```
[the remark this answers is not in the document]
Comment (gabezurita) [reply src/applications/accredited-representative-portal/components/LandingPageWelcome/LandingPageWelcome.jsx]: Had the change stashed! https://github.com/department-of-veterans-affairs/vets-website/pull/30146/commits/0c7cc4985fbdac5b6c7b4d5d37342c76f1229f89

**Review by williamphelps13 — APPROVED** [review]
This is a huge step forward for ARP. Great job, Gabe!

**Comment by gabezurita** [comment]
> This is a huge step forward for ARP. Great job, Gabe!

Thanks, @williamphelps13 for the thorough 🦅 👁️ review!

**Pull request merged**